v2.2/architecture/life-of-a-distributed-transaction.md

```diff
@@ -0,0 +1,204 @@
+
+The leaseholder then proposes these Raft operations to the Raft group leader. Most often, the leaseholder and the Raft leader are the same node, but there are situations where the roles might drift to different nodes. However, when the two roles are not collocated on the same physical machine, CockroachDB will attempt to relocate them on the same node at the next opportunity.
+
+## Raft Leader
+
+CockroachDB leverages Raft as its consensus protocol. If you aren't familiar with it, we recommend checking out the details about [how CockroachDB leverages Raft](https://www.cockroachlabs.com/docs/v2.1/architecture/replication-layer.html#raft), as well as [learn more about how the protocol works at large](http://thesecretlivesofdata.com/raft/).
+
+In terms of executing transactions, the Raft leader receives proposed Raft commands from the leaseholder. Each Raft command is a write that is used to represent an atomic state change of the underlying key-value pairs stored in RocksDB.
+
+### Consensus
+
+For each command the Raft leader receives, it proposes a vote to the other members of the Raft group.
+
+Once it achieves consensus (i.e. a majority of nodes including itself acknowledge the Raft command), it is committed to the Raft leader’s Raft log, and written to RocksDB. The other nodes move their Raft log ahead as soon the Raft leader indicates the group reached consensus on the write.
+
+Once the leader commits the Raft log entry, it’s considered committed. At this point the value is considered written, and if another operation comes in and performs a read on RocksDB for this key, they’ll encounter this value.
+
+Note that this write operation creates a write intent; these writes will not be fully committed until the gateway node's `TxnCoordSender` sets the transaction record's status to `COMMITTED`.
+
+## On the way back up
+
+Now that we have followed an operation all the way down from the SQL client to RocksDB, we can pretty quickly cover what happens on the way back up (i.e., generating a response to the client).
+
+1. Once the leaseholder applies a write to its Raft log, it sends an commit acknowledgment to the gateway node's `DistSender`, which was waiting for this signal (having already received the provisional acknowledgment from the leaseholder's evaluator).
+1. The gateway node's `DistSender` aggregates commit acknowledgments from all of the write operations in the `BatchRequest`, as well as any values from read operations that should be returned to the client.
+1. Once all operations have successfully completed, the `DistSender` sends a response to the `TxnCoordSender`, indicating that the transaction completed.
+1. The `TxnCoordSender` checks the transaction record. There are a few situations that might arise here:
```

Comment:
Also, note that the `TxnCoordSender` never does anything itself. It always operates by sending `BatchRequests` down through a `DistSender`.